### PR TITLE
[cni-cilium] unprivileged mode

### PR DESCRIPTION
**Pull request by fl64**
## Description
Run CNI cilium in a non-privileged environment with the maximum permissions restriction.

## Why do we need it, and what problem does it solve?
Reduce the attack surface
Closes (https://github.com/deckhouse/deckhouse/issues/2961)

## What is the expected result?
The cilium agent runs in an unprivileged environment

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cni-cilium
type: chore
summary: Run CNI cilium in a non-privileged environment with the maximum permissions restriction.
impact: All cilium pods will be restarted
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @fl64. I've started the workflow for label 'e2e/run/openstack'...

File: modules/021-cni-cilium/templates/agent/daemonset.yaml [hunk 1]
```diff
@@ -30,6 +30,9 @@ spec:
       annotations:
         configmap-checksum: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum | quote }}
         scheduler.alpha.kubernetes.io/critical-pod: ""
+        container.apparmor.security.beta.kubernetes.io/cilium-agent: "unconfined"
+        container.apparmor.security.beta.kubernetes.io/clean-cilium-state: "unconfined"
+        container.apparmor.security.beta.kubernetes.io/mount-cgroup: "unconfined"
```
Comment (RomanenkoDenys) [inline modules/021-cni-cilium/templates/agent/daemonset.yaml:35]: is this amnnotations properly work on Centos based distros ?

**Review by RomanenkoDenys — COMMENTED** [review]
some questions

**Review by zuzzas — APPROVED** [review]

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Pull request merged**